### ISSUE TITLE
Some common diacritical marks are missing

### ISSUE THREAD

**Issue by ibnesayeed** [saadatm/navees#1]
I noticed that "do zer" and "do pesh" are not mapped in this keyboard yet.

**Comment by saadatm** [comment]
Yes. These marks were missing in the original CRULP layout as well, most probably because they are not used in Urdu... or are they? :-)

**Comment by ibnesayeed** [comment]
I don't remember any word on top of my head where these were used, but I do remember a few occasions when I needed them while writing something in Urdu Mehfil. Most probably, those were Arabic words that were needed in the text (which is not uncommon). I have seen mention of "tanveen" in Eijaz Chachu's Urdu Sauti Kaleedi Takhta. Hence, for the sake of completeness they should be added, though, it is fine to keep them behind some modifier key combo if necessary.

**Comment by mahnoorfirdous** [comment]
Let's be honest, even if you add tanween, which some of them do have, you'd need mada and different positions of hamza and sakoon...I've seen some that come close but then they omitted the "hamza" + "ye" combo to add the "barhi pesh" and "Khari zabar". So you might end up using two or three keyboards at the same time XD

**Comment by saadatm** [comment]
@mahnoorfirdous Not sure what you mean; we already have _tanween_ (دو زبر), _sukoon_, _ulta pesh_ and _kharra zabar_ in Navees. Also, how is the addition of tanween related to "mada and different positions of hamza and sakoon"?

**Comment by mahnoorfirdous** [comment]
ئ ئی۔ For example I can't get these two characters on a keyboard where I have the pesh (wherever it's mapped, I can't find it and I can't see it on the listed mapping for CPRULP layout so I switched to a different one but now I can't get these two characters I've just written here). And by the madaa I meant the Arabic madaa. And the different positions of hamza that you need to type words like this 
![image](https://user-images.githubusercontent.com/88815964/175821693-3a345e9f-4b0a-4c4d-93c0-86fb6d49f676.png)
The reason I wanted a keyboard with harakaat was to type Arabic. But it's still very good at what it does, your version. These are nitpicks.

**Comment by saadatm** [comment]
Ah, I see. There are multiple things here:

1. If you are typing Arabic, I would strongly suggest using an Arabic keyboard, which will not only have all the _harakat_, but will also have the correct Unicode characters. (Some characters that are common in Arabic and Urdu have different code points in Unicode; for instance, ه in Arabic (U+0647) is different from ہ in Urdu (U+06C1), and an Urdu keyboard is unlikely to have the correct code points for Arabic. [Here's a table](https://files.saadatmand.pk/type/similar-arabic-chars.html) that lists some such characters).
2. Unicode text of the Quran is quite nuanced, and I usually prefer to copy it from a reliable source that has already sorted out all or most of its complexities. Here's one: https://github.com/aliftype/quran-data
3. Again for the Quran, you will also need a font that contains all the needed glyphs and rules necessary for a good rendering. Below is the same _ayat_ that you posted in your image; I copied its text from the above mentioned source, and set it in the [Amiri Quran](https://www.amirifont.org/) font:
![image](https://user-images.githubusercontent.com/6615143/175864816-dcf83382-f7f3-4219-a3e9-f4fc4407d7e7.png)
